example_client/grpc_stateful_client.py

```diff
@@ -47,38 +85,63 @@
 print('Reading ark file {}'.format(model_input_path))
 ark_reader = ArchiveReader(model_input_path)
 
+model_score_path = args.get('model_score_path')
+print('Reading scores ark file {}'.format(model_score_path))
+ark_score = ArchiveReader(model_score_path)
+numberOfKeys = 0
+
 for key, obj in ark_reader:
-    print("Input ark file data range {0}: {1}".format(key, obj.shape))
+    printDebug("Input ark file data range {0}: {1}".format(key, obj.shape))
+    numberOfKeys += 1
+
+for key, obj in ark_score:
+    printDebug("Scores ark file data range {0}: {1}".format(key, obj.shape))
+
+scoreObjects = { k:m for k,m in ark_score }
 
 print('Start processing:')
 print('\tModel name: {}'.format(args.get('model_name')))
 
 SEQUENCE_START = 1
 SEQUENCE_END = 2
+sequence_id = 1020
+utterances_limit = int(args.get('utterances'))
+utterance = 0
 
-sequence_id = 1001
+meanErrGlobal = 0.0
 for key, obj in ark_reader:
+    utterance += 1
+    if utterance > utterances_limit:
+        break
+
     batch_size = obj.shape[0]
-    print('\tInput name: {}\n'.format(key))
-    print('\tInput in shape: {}\n'.format(obj.shape))
-    print('\tInput batch size: {}\n'.format(batch_size))
-    print('\tSequence id: {}\n'.format(sequence_id))
+    print('\n\tInput name: {}'.format(key))
+    printDebug('\tInput in shape: {}'.format(obj.shape))
+    printDebug('\tInput batch size: {}'.format(batch_size))
+    printDebug('\tSequence id: {}'.format(sequence_id))
+
+    meanErrSum = 0.0
+
+    samples_limit = int(args.get('samples'))
     for x in range(0, batch_size):
 
-        print('\tExecution: {}\n'.format(x))
+        if x > samples_limit:
+            break
+
+        printDebug('\tExecution: {}\n'.format(x))
         request = predict_pb2.PredictRequest()
         request.model_spec.name = args.get('model_name')
 
-        print('\tTensor before input in shape: {}\n'.format(obj[x].shape))
-        print('\tTensor input in shape: {}\n'.format(expand_dims(obj[x], axis=0).shape))
+        printDebug('\tTensor before input in shape: {}\n'.format(obj[x].shape))
+        printDebug('\tTensor input in shape: {}\n'.format(expand_dims(obj[x], axis=0).shape))
 
         request.inputs[args['input_name']].CopyFrom(make_tensor_proto(obj[x], shape=(expand_dims(obj[x], axis=0).shape)))
         if x == 0:
             request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
         
         request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
 
-        if x == batch_size - 1:
+        if x == batch_size - 1 or x == samples_limit:
```

Comment:
If you apply above suggestion with:
```
if x >= samples_limit:
    break
```
Then you would need to change this one to: `or x == samples_limit - 1`